mat/vector_test.go

```diff
@@ -51,6 +51,40 @@ func TestNewVector(t *testing.T) {
 	}
 }
 
+func TestCap(t *testing.T) {
+	for i, test := range []struct {
+		vector *Vector
+		want   int
+	}{
+		{vector: NewVector(3, nil), want: 3},
+		{
```

Comment:
I forget, do we guarantee that the last element in the vector is valid? If not, could you add a test for all of the remainders? That is, check that 10, 11 and 12 all give a capacity of 4?